resources/dw/resource_hive_vw.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type hiveResource struct {
+	client *cdp.Client
+}
+
+var (
+	_ resource.Resource              = (*hiveResource)(nil)
+	_ resource.ResourceWithConfigure = (*hiveResource)(nil)
+)
+
+func NewHiveResource() resource.Resource {
+	return &hiveResource{}
+}
+
+func (r *hiveResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *hiveResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vw_hive"
+}
+
+func (r *hiveResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = hiveSchema
+}
+
+func (r *hiveResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var plan hiveResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generate API request body from plan
+	vw := operations.NewCreateVwParamsWithContext(ctx).
+		WithInput(&models.CreateVwRequest{
+			Name:      plan.Name.ValueStringPointer(),
+			ClusterID: plan.ClusterID.ValueStringPointer(),
+			DbcID:     plan.DbCatalogID.ValueStringPointer(),
+			VwType:    models.VwTypeHive.Pointer(),
+		})
+
+	// Create new virtual warehouse
+	response, err := r.client.Dw.Operations.CreateVw(vw)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating hive virtual warehouse",
+			"Could not create hive, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	payload := response.GetPayload()
+	desc := operations.NewDescribeVwParamsWithContext(ctx).
+		WithInput(&models.DescribeVwRequest{VwID: &payload.VwID, ClusterID: plan.ClusterID.ValueStringPointer()})
+	describe, err := r.client.Dw.Operations.DescribeVw(desc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating hive virtual warehouse",
+			"Could not describe hive, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	hive := describe.GetPayload()
+
+	// Map response body to schema and populate Computed attribute values
+	plan.ID = types.StringValue(hive.Vw.ID)
+	plan.DbCatalogID = types.StringValue(hive.Vw.DbcID)
+	plan.Name = types.StringValue(hive.Vw.Name)
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *hiveResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	//TODO implement me
```

Comment:
if this `tflog.Warn` shall be the "implementation" then this comment can be removed